velox/exec/HashProbe.h

```diff
@@ -108,11 +159,96 @@ class HashProbe : public Operator {
 
   void ensureLoadedIfNotAtEnd(column_index_t channel);
 
+  // Indicates if the operator has more probe inputs from either the upstream
+  // operator or the spill input reader.
+  bool hasMoreInput() const;
+
+  // Indicates if the join type such as right, right semi and full joins,
+  // require to produce the output results from the join results after all the
+  // probe operators have finished processing the probe inputs.
+  bool needLastProbe() const;
+
+  // Indicates if the join type can skip processing probe inputs with empty
+  // build table such as inner, right and semi joins.
+  bool skipProbeOnEmptyBuild() const;
+
+  bool spillEnabled() const;
+
+  // Indicates if the probe input is read from spilled data or not.
+  bool isSpillInput() const;
+
+  // Indicates if there is more spill data to restore after finishes processing
+  // the current probe inputs.
+  bool hasMoreSpillData() const;
+
+  // Indicates if the operator needs to spill probe inputs. It is true if parts
+  // of the build-side rows have been spilled. Hence, the probe operator needs
+  // to spill the corresponding probe-side rows as well.
+  bool needSpillInput() const;
+
+  // Invoked after finishes processing the probe inputs and there is spill data
+  // remaining to restore. The function will reset the internal states which
+  // are relevant to the last finished probe run. The last finished probe
+  // operator will also notify the hash build operators to build the next hash
+  // table from spilled data.
+  void prepareForSpillRestore();
+
+  // Invoked to read next batch of spilled probe inputs from disk to process.
+  void addSpillInput();
+
+  // Invoked to spill rows in 'input' to disk directly if the corresponding
```

Comment:
Perhaps, clarify that this method keeps 'input' as is if no row needs spilling; resets it to null if all rows need spilling; wraps in a dictionary using rows number that do need spilling otherwise.

It is a bit unusual to wrap the top-level row vector in a dictionary.